Switch to using go modules

mem opened the pull request:
- Add go.mod (and the corresponding go.sum)
- Fix makeit/gengodep so that it lists all the go files that are
  needed to build a particular import path
- Add internal/pkg/runtime/engines/singularity/plugins.go to have
  something that pulls the relevant plugin code.
- Remove Gopkg.toml and Gopkg.lock
- Updated vendor/ with go mod vendor
- Make sure we are calling the same go compiler in all cases by
  resolving the absolute path only once.
- Explicitly export GO111MODULE=on to make sure that everything is
  running with module support activated. In particular, this seems
  necessary to build the CNI plugins, as having a go.mod file and
  passing -mod=vendor in that case doesn't seem to be sufficient (go
  still fails to locate the relevant modules).
- Replace dep with go mod

Signed-off-by: Michael Bauer <bauerm@umich.edu>
Signed-off-by: Marcelo E. Magallon <marcelo@sylabs.io>

bauerm97 committed:
Switch to using go modules

- Add go.mod (and the corresponding go.sum)
- Fix makeit/gengodep so that it lists all the go files that are
  needed to build a particular import path
- Add internal/pkg/runtime/engines/singularity/plugins.go to have
  something that pulls the relevant plugin code.
- Remove Gopkg.toml and Gopkg.lock
- Updated vendor/ with go mod vendor
- Make sure we are calling the same go compiler in all cases by
  resolving the absolute path only once.
- Explicitly export GO111MODULE=on to make sure that everything is
  running with module support activated. In particular, this seems
  necessary to build the CNI plugins, as having a go.mod file and
  passing -mod=vendor in that case doesn't seem to be sufficient (go
  still fails to locate the relevant modules).
- Replace dep with go mod

Signed-off-by: Michael Bauer <bauerm@umich.edu>
Signed-off-by: Marcelo E. Magallon <marcelo@sylabs.io>

cclerget commented on mlocal/frags/build_network.mk:
```diff
@@ -57,12 +58,12 @@ cni_config_INSTALL := $(DESTDIR)$(SYSCONFDIR)/singularity/network
 		if [ ! -f $$cniplugin ]; then \
 			echo " CNI PLUGIN" $$name; \
-		go build $(GO_BUILDMODE) -tags "$(GO_TAGS)" $(GO_LDFLAGS) $(GO_GCFLAGS) $(GO_ASMFLAGS) \
+		go build $(GO_MODFLAGS) $(GO_BUILDMODE) -tags "$(GO_TAGS)" $(GO_LDFLAGS) $(GO_GCFLAGS) $(GO_ASMFLAGS) \
```
```suggestion
		GOPATH=$(SOURCEDIR)/vendor go build $(GO_MODFLAGS) $(GO_BUILDMODE) -tags "$(GO_TAGS)" $(GO_LDFLAGS) $(GO_GCFLAGS) $(GO_ASMFLAGS) \
```

mem replied:
Thanks! I'm going to guess you ran into the same issue I did, where sometimes the CNI plugins couldn't be found.

Experimenting around, setting GO111MODULE=on explicitly seems to take care of this in all cases.

cclerget commented on internal/pkg/runtime/engines/singularity/plugins.go:
```diff
@@ -0,0 +1,25 @@
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+// +build cni_plugins
```
To avoid confusion, maybe add a comment mentioning this is a hack for `go mod` and `cni_plugins` build tag is not intended to be used during build

mem replied:
Good point, I'll update this.

cclerget reviewed (changes requested):
Also tested by removing vendor and executing `go mod vendor`, everything goes back without any issues. Nice work !

Commit:
Clean up building of CNI plugins

- Remove some redundant variables and compute values from original list.
- Document the purpose and usage of
  internal/pkg/runtime/engines/singularity/plugins.go

Signed-off-by: Marcelo E. Magallon <marcelo@sylabs.io>

Commit:
Update Go to 1.11.6

- Change Circle CI configuration to require Go 1.11.6
- Set required Go version to 1.11
- Remove checks that use version 1.10

Signed-off-by: Marcelo E. Magallon <marcelo@sylabs.io>

Commit:
Up build verbosity in CircleCI

Signed-off-by: Marcelo E. Magallon <marcelo@sylabs.io>
